File: 26_02_2022/Longest_ordered_vowel_sequence.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

bool isCorrect(string s){
    string vovels = "aeiou";
    int ptr = 0;
    for(int i=0;i<s.length();i++){
        if(ptr == 0){
            if(vovels.find(s[i]) != string::npos){
                ptr = vovels.find(s[i]);
            }
        }
        else{
            if(vovels.find(s[i]) != string::npos){
                if(vovels.find(s[i]) < ptr){
                    return false;
                }
                else{
                    ptr = vovels.find(s[i]);
                }
            }
        }
    }
    return true;
}

int maxx(int a,int b){
    if(a>b){
        return a;
    }
    else{
        return b;
    }
}

void maxx_unique(vector<string> words){
    string vovels = "aeiou";
    int maxx = 0;
    vector<string> data;
    for(int i=0;i<words.size();i++){
        int count = 0;
        string x = "";
        for(int j=0;j<words[i].size();j++){
            if(vovels.find(words[i][j]) != string::npos){
                if(x.find(words[i][j]) == string::npos){
                    x += words[i][j];
                    count++;
                }
            }
        }
        if(count>maxx){
            data.clear();
            data.push_back(words[i]);
            maxx = count;
        }
        else if(count == maxx){
            data.push_back(words[i]);
        }
    }

    for(int i=0;i<data.size();i++){
        cout<<data[i]<<endl;
    }

}

int main(){


    // getting the string and splitting into spaces and converting it to words
    string s;
    getline (cin, s);

    vector<string> words;

    string word = "";
    for(auto i:s){
        if(i != ' '){
            word += i;
        }
        else{
            words.push_back(word);
            word = "";
        }
    }

    // finding all the words that are correct

    vector<string> correct_words;
    for(int i=0;i<words.size();i++){
        if(isCorrect(words[i])){
            correct_words.push_back(words[i]);
        }
    }

    // finding the max one and printing its size
    maxx_unique(correct_words);

    return 0;
}

File: 26_02_2022/Longest_ordered_vowel_sequence.py
def isCorrect(n):
    vowels = 'aeiou'
    ptr = 0
    for i in range(len(n)):
        if(ptr==0):
            if n[i] in vowels:
                ptr=vowels.index(n[i])
        else:
            if n[i] in vowels:
                if vowels.index(n[i])<ptr:
                    return False
                else:
                    ptr=vowels.index(n[i])
    return True


def max_unique_vovels(words):
    vovels = 'aeiou'
    maxx = 0
    final = []
    for i in words:
        count = 0
        x = ''
        for j in i:
            if(j in vovels and j not in x):
                x += j
                count += 1
        if(count>maxx):
            final = [i]
            maxx = count
        elif(count == maxx):
            final.append(i)
    return final
n = input().split()
words = []

for i in n:
    if(isCorrect(i)):
        words.append(i)

maxx_unique = max_unique_vovels(words)
for i in maxx_unique:
    print(i)